LightBlue/Hosted/HostedAzureBlockBlob.cs

```diff
@@ -1,30 +1,47 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Linq;
+using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.WindowsAzure.Storage;
-using Microsoft.WindowsAzure.Storage.Auth;
-using Microsoft.WindowsAzure.Storage.Blob;
+using Azure;
+using Azure.Storage;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+using Azure.Storage.Sas;
 
 namespace LightBlue.Hosted
 {
     public class HostedAzureBlockBlob : IAzureBlockBlob
     {
-        private readonly CloudBlockBlob _cloudBlockBlob;
+        private readonly BlockBlobClient _cloudBlockBlob;
+        private BlobProperties _properties;
 
-        public HostedAzureBlockBlob(CloudBlockBlob cloudBlockBlob)
+        public HostedAzureBlockBlob(BlockBlobClient cloudBlockBlob, IDictionary<string, string> metadata = null)
         {
             _cloudBlockBlob = cloudBlockBlob;
+
+            if (metadata?.Any() == true)
+            {
+                _properties = new BlobProperties();
+                foreach (var m in metadata)
+                    _properties.Metadata[m.Key] = m.Value;
+            }
         }
 
         public HostedAzureBlockBlob(Uri blobUri)
         {
-            _cloudBlockBlob = new CloudBlockBlob(blobUri);
+            _cloudBlockBlob = new BlockBlobClient(blobUri);
```

Comment:
Setting of _properties is optional in the ctors here, but Public Properties `CopyState` and `Metadata` will hit a null reference when being used in that scenario.  Mercury for example when saving Blobs will hit this issue.